### ISSUE TITLE
`zetacore` : update protocol buffers 

### ISSUE THREAD

**Issue by kingpinXD** [zeta-chain/node#2693]
Original trail of bits issue
![image](https://github.com/user-attachments/assets/82ba9105-e800-49fa-8e56-a7da8704bfac)

**Comment by kingpinXD** [comment]
The main reason for this change is to move away from `github.com/gogo/protobuf`, which is not maintained, and use `github.com/cosmos/gogoproto` instead.

The upgrade was not done during the migration to 0.47 because of the following reason 

- We use go-tss , which is a fork of the original library maintained by zetachain
https://github.com/zeta-chain/go-tss

- However go-tss has external dependencies which use `github.com/gogo/protobuf`

Used by go-tss -> [github.com/libp2p/go-libp2p-kad-dht](http://github.com/libp2p/go-libp2p-kad-dht)
https://github.com/libp2p/go-libp2p-kad-dht/blob/25d299e64f3cb9a3da337e9d4af1a1b722ae503d/go.mod#L8

Used by go-tss -> [github.com/cockroachdb/errors/errbase](http://github.com/cockroachdb/errors/errbase)
https://github.com/cockroachdb/errors/blob/5b23426aab2f13826e734bad3ffa04568ee03f62/go.mod#L11



This makes it difficult for us to make the change as requested.
The node and go-tss repos , do not use `github.com/gogo/protobuf` and the import is indirect in both cases.

I don't think it's a major issue for us right now.

**Comment by gartnera** [comment]
> The upgrade was not done during the migration to 0.47 because of the following reason

We did move to github.com/cosmos/gogoproto during the 0.47 upgrade. But go-tss still depends on `github.com/gogo/protobuf`.

**Comment by kingpinXD** [comment]
> > The upgrade was not done during the migration to 0.47 because of the following reason
> 
> We did move to `github.com/cosmos/gogoproto` during the 0.47 upgrade. Everything on the cosmos side uses `github.com/cosmos/gogoproto`. But go-tss still depends on `github.com/gogo/protobuf`.

Ya, that's exactly what I said, 
I don't think the indirect dependency is a cause for concern, do you feel otherwise?

**Comment by gartnera** [comment]
Yeah I'm not particularly concerned about it. Looks like there some work in progress to remove it from the libp2p side.
